### ISSUE TITLE
A postgresql Interval in a UDT fails to be converted correctly

### ISSUE THREAD

**Issue by jason-adnuntius** [jOOQ/jOOQ#11869]
### Expected behavior

I would expect an interval to be converted correctly from a interval field to a YearInSecond java field.   

### Actual behavior

I receive a UnsupportedOperationException instead:

```
Caused by: java.lang.UnsupportedOperationException: Class class org.jooq.types.YearToSecond is not supported
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.pgFromString(DefaultBinding.java:3391)
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.pgSetValue(DefaultBinding.java:3440)
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.access$200(DefaultBinding.java:3182)
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding$1.operate(DefaultBinding.java:3432)
	at org.jooq.impl.RecordDelegate.operate(RecordDelegate.java:130)
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.pgNewRecord(DefaultBinding.java:3425)
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.get0(DefaultBinding.java:3261)
	at org.jooq.impl.DefaultBinding$DefaultRecordBinding.get0(DefaultBinding.java:3182)
	at org.jooq.impl.DefaultBinding$AbstractBinding.get(DefaultBinding.java:824)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.setValue(CursorImpl.java:1725)
	... 47 more
```

### Steps to reproduce the problem

Ive included a reproducible test case, for both a UDT interval field in an array but also directly, both fail with the unsupported operation exception.

https://github.com/jason-adnuntius/jOOQ-mcve

### Versions

The mcve project has all the information regarding versions, please note Ive tried this with 3.14.4 as well as 2.12.4

**Comment by jason-adnuntius** [comment]
Is there a plan to support?   Because in 3.11 it was possible to support them with a custom binding.

**Comment by lukaseder** [comment]
I'm sorry for the delay. This may have worked before by accident. We certainly don't have an integration test yet that combines these features. This will be fixed eventually, but I can't promise any deadlines.

> Because in 3.11 it was possible to support them with a custom binding

Can you show an example of such a binding that worked and stopped working?

**Comment by jason-adnuntius** [comment]
Hi,

We declared the field thus:

public static final UDTField<ObjectiveDeliveryPredictionRecord, Duration> TIME_REQUIRED = createField("time_required", org.jooq.impl.DefaultDataType.getDefaultDataType("\"pg_catalog\".\"interval\""), OBJECTIVE_DELIVERY_PREDICTION, "", new JooqIntervalBinding());

And attached are the relevant binding classes

[Jooq.zip](https://github.com/jOOQ/jOOQ/files/6746157/Jooq.zip)

**Comment by LightSystem** [comment]
I've encountered this same exception while using `select(TABLE, ...)` where `TABLE` has a column of type `interval`. I'm assuming this is because of the use of `row()` for getting a record of `TABLE`.
What are my options here? Am I forced to explode the columns of `TABLE` and get them individually?

**Comment by lukaseder** [comment]
Sorry, @LightSystem, I can't tell you how to best write your query right now, given that there's a feature combination that hasn't yet been covered by integration tests.

**Comment by lukaseder** [comment]
Duplicate of #14437, fixed in jOOQ 3.18.0, 3.17.9. That issue is about embedding `INTERVAL` types in `MULTISET`, but the logic of deserialising PostgreSQL's object notation is shared, so the fix is the same.
